### PR TITLE
[SPARK-6917] [SQL] DecimalType is not read back when non-native type exists

### PR DESCRIPTION
**Pull request by davies**
cc @yhuai

### CODE REVIEW

**Comment by JoshRosen** [comment]
Regression test?

**Comment by yhuai** [comment]
Yeah, let's add a test.

**Comment by davies** [comment]
regression test added.

File: sql/core/src/main/scala/org/apache/spark/sql/parquet/ParquetConverter.scala [hunk 1]
```diff
@@ -244,7 +244,7 @@ private[parquet] abstract class CatalystConverter extends GroupConverter {
    * Read a decimal value from a Parquet Binary into "dest". Only supports decimals that fit in
    * a long (i.e. precision <= 18)
    */
-  protected[parquet] def readDecimal(dest: Decimal, value: Binary, ctype: DecimalType): Unit = {
+  protected[parquet] def readDecimal(dest: Decimal, value: Binary, ctype: DecimalType): Decimal = {
```
Comment (yhuai) [inline sql/core/src/main/scala/org/apache/spark/sql/parquet/ParquetConverter.scala]: oh, I just realized that when we reuse the Decimal value, we do not really need to use the returned value. But, we have another place that needs the returned value. Can we add a comment at here?

**Comment by davies** [comment]
@yhuai fixed.

**Pull request closed without merging**